### PR TITLE
(LD-472) Add local storage cache for API requests

### PR DESCRIPTION
**Pull request by snoattoh**
This PR adds cache for API requests

- updates storeData to return the setItem instead of awaiting it
- updates apiRequest helper to maintain a cache for each unique request for one minute.

### CODE REVIEW

File: src/services/storage.ts [hunk 1]
```diff
@@ -6,10 +6,7 @@ export const getData = async <T>(location: string): Promise<T | undefined> => {
...
 }
 
-export const storeData = async <T>(
-  location: string,
-  value: T,
-): Promise<void> => {
+export const storeData = <T>(location: string, value: T): Promise<void> => {
   const jsonValue = JSON.stringify(value)
-  await AsyncStorage.setItem(location, jsonValue)
+  return AsyncStorage.setItem(location, jsonValue)
```
Comment (chasenlehara) [inline src/services/storage.ts]: Let’s change `location` to `key`.

File: src/services/pmo/api/api.ts [hunk 2]
```diff
@@ -17,6 +24,21 @@ export async function apiRequest<
 }): Promise<{ data: Data | null; error: Error | null }> {
   try {
     const query = params ? stringifyQuery(params) : ""
+    const cachedResponse = await getData<LocalStorageApiRequest<Data>>(
+      `${baseUrl}${path}?${query}`,
```
Comment (chasenlehara) [inline src/services/pmo/api/api.ts]: This gets concatenated in two places. I would store it in a variable.

File: src/services/pmo/api/api.ts [hunk 3]
```diff
@@ -30,6 +52,13 @@ export async function apiRequest<
       ? null
       : new Error(`${response.status} (${response.statusText})`)
 
+    if (method === "GET" && response.ok) {
+      storeData<LocalStorageApiRequest<Data>>(`${baseUrl}${path}?${query}`, {
+        data: data,
+        dateTime: new Date(),
```
Comment (chasenlehara) [inline src/services/pmo/api/api.ts:58]: I would consider storing `new Date().valueOf()` so you don’t have to create a `Date` object above.

File: src/services/pmo/api/api.ts [hunk 4]
```diff
@@ -17,6 +24,21 @@ export async function apiRequest<
 }): Promise<{ data: Data | null; error: Error | null }> {
   try {
     const query = params ? stringifyQuery(params) : ""
+    const cachedResponse = await getData<LocalStorageApiRequest<Data>>(
+      `${baseUrl}${path}?${query}`,
+    )
+    if (cachedResponse) {
+      const currentTime = Math.abs(
```
Comment (chasenlehara) [inline src/services/pmo/api/api.ts]: `Math.abs` shouldn’t be needed.

File: src/services/pmo/api/api.ts [hunk 5]
```diff
@@ -17,6 +24,21 @@ export async function apiRequest<
...
+        new Date().valueOf() - new Date(cachedResponse.dateTime).valueOf(),
+      )
+      if (currentTime < 60000) {
```
Comment (chasenlehara) [inline src/services/pmo/api/api.ts]: I would put a comment next to the 60000 so it’s clear that this is one minute.

**Review by chasenlehara — CHANGES_REQUESTED** [review]

File: src/services/storage.ts [hunk 6]
```diff
@@ -6,10 +6,7 @@ export const getData = async <T>(location: string): Promise<T | undefined> => {
   return jsonValue
 }
 
-export const storeData = async <T>(
```
Comment (christopherjbaker) [inline src/services/storage.ts:9]: While you're changing this file, would you convert it to a modlet?

**Review by chasenlehara — APPROVED** [review]

**Pull request merged**